### ISSUE TITLE
ansible fails to login using ssh: Permission denied (publickey) error

### ISSUE THREAD

**Issue by InfiniteCoder** [sovereign/sovereign#736]
When running ansible, I get the error 

    fatal: [1.1.1.1]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: deploy@1.1.1.1: Permission denied (publickey).\r\n", "unreachable": true}

The server is hosted on DigitalOcean. I have copied the ssh key used for root user to deploy user, and verified that this works by ssh-ing to deploy user via terminal.

I think the error must be that either ansible doesn't use a ssh key at all, or it tries to use wrong ssh key. There are multiple ssh keys in my `~/.ssh/` directory, and the correct one is NOT the default one, i.e id_rsa.

Ansible version details are as follows,

```
$ ansible --version
ansible 2.4.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/aditya/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.14 (default, Jan  5 2018, 10:41:29) [GCC 7.2.1 20171224]
```

I have also run the ansible command using -vvvv parameter, but I'm not sure if it's safe to paste that dump here, in case it contains sensitive information.

**Comment by neuhaus** [comment]
See https://serverfault.com/questions/720326/ansible-with-ssh-keys

you have to set `private_key_file`

**Comment by jonstump** [comment]
I've set the private key file and I still get errors. Mine is very similar to the above only i get permission denied because of 'invalid format'

```
fatal: [1.1.1.1]: UNREACHABLE! => {
    "changed": false,
    "msg": "Failed to connect to the host via ssh: Load key \"/home/USER/.ssh/id_rsa.pub\": invalid format\r\ndeploy@45.33.41.8: Permission denied (publickey,password).",
    "unreachable": true
```
    
 I've tried multiple things without luck. I kind of feel like I'm running in circles on this to get error messages that just lead me back to this one.

EDIT: looks like my issue was a rogue ansible config file with nothing in it in my user home director
